File: back/src/main/java/com/back/domain/member/dto/AdminMemberDto.java
package com.back.domain.member.dto;

import com.back.domain.member.entity.Member;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDateTime;

public record AdminMemberDto(
        @NotNull
        Long id,
        @NotNull
        String username,
        String nickname,
        String githubId,
        @NotNull
        Boolean isAdmin,
        @NotNull
        Boolean isDeleted,
        @NotNull
        LocalDateTime createdDate
) {
    public AdminMemberDto(Member member) {
        this(
                member.getId(),
                member.getUsername(),
                member.getNickname(),
                member.getGithubId(),
                member.isAdmin(),
                member.isDeleted(),
                member.getCreatedDate()
        );
    }
}

File: back/src/main/java/com/back/domain/member/dto/AdminMemberDto.kt
package com.back.domain.member.dto

import com.back.domain.member.entity.Member
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime

class AdminMemberDto(
    id: Long,
    githubId: String?,
    githubLink: String?,
    val username: String?,
    val nickname: String?,
    @field:NotNull val isAdmin: Boolean,
    @field:NotNull val isDeleted: Boolean,
    @field:NotNull val createdDate: LocalDateTime
) : MemberDto(id, githubId, githubLink) {
    constructor(member: Member) : this(
        member.getId(),
        member.githubId,
        member.githubLink,
        member.username,
        member.nickname,
        member.isAdmin,
        member.isDeleted,
        member.getCreatedDate()
    )
}